project-base/app/src/DataFixtures/Demo/WatchdogDataFixture.php

```diff
@@ -0,0 +1,159 @@
+                self::ATTRIBUTE_PRODUCT_KEY => '6',
+            ],
+            [
+                self::ATTRIBUTE_DOMAIN_KEY => Domain::SECOND_DOMAIN_ID,
+                self::ATTRIBUTE_EMAIL_KEY => 'charlie.white@example.com',
+                self::ATTRIBUTE_PRODUCT_KEY => '7',
+            ],
+            [
+                self::ATTRIBUTE_DOMAIN_KEY => Domain::FIRST_DOMAIN_ID,
+                self::ATTRIBUTE_EMAIL_KEY => 'lucy.adams@example.com',
+                self::ATTRIBUTE_PRODUCT_KEY => '8',
+            ],
+            [
+                self::ATTRIBUTE_DOMAIN_KEY => Domain::SECOND_DOMAIN_ID,
+                self::ATTRIBUTE_EMAIL_KEY => 'george.evans@example.com',
+                self::ATTRIBUTE_PRODUCT_KEY => '1',
+            ],
+            [
+                self::ATTRIBUTE_DOMAIN_KEY => Domain::FIRST_DOMAIN_ID,
+                self::ATTRIBUTE_EMAIL_KEY => 'olivia.martin@example.com',
+                self::ATTRIBUTE_PRODUCT_KEY => '10',
+            ],
+            [
+                self::ATTRIBUTE_DOMAIN_KEY => Domain::SECOND_DOMAIN_ID,
+                self::ATTRIBUTE_EMAIL_KEY => 'james.wilson@example.com',
+                self::ATTRIBUTE_PRODUCT_KEY => '11',
+            ],
+            [
+                self::ATTRIBUTE_DOMAIN_KEY => Domain::FIRST_DOMAIN_ID,
+                self::ATTRIBUTE_EMAIL_KEY => 'sophia.moore@example.com',
+                self::ATTRIBUTE_PRODUCT_KEY => '12',
+            ],
+            [
+                self::ATTRIBUTE_DOMAIN_KEY => Domain::SECOND_DOMAIN_ID,
+                self::ATTRIBUTE_EMAIL_KEY => 'henry.jackson@example.com',
+                self::ATTRIBUTE_PRODUCT_KEY => '13',
+            ],
+            [
+                self::ATTRIBUTE_DOMAIN_KEY => Domain::FIRST_DOMAIN_ID,
+                self::ATTRIBUTE_EMAIL_KEY => 'mia.thomas@example.com',
+                self::ATTRIBUTE_PRODUCT_KEY => '14',
+            ],
+            [
+                self::ATTRIBUTE_DOMAIN_KEY => Domain::SECOND_DOMAIN_ID,
+                self::ATTRIBUTE_EMAIL_KEY => 'liam.harris@example.com',
+                self::ATTRIBUTE_PRODUCT_KEY => '15',
+            ],
+            [
+                self::ATTRIBUTE_DOMAIN_KEY => Domain::FIRST_DOMAIN_ID,
+                self::ATTRIBUTE_EMAIL_KEY => 'amelia.clark@example.com',
+                self::ATTRIBUTE_PRODUCT_KEY => '16',
+            ],
+        ];
+    }
+
+    /**
+     * @param array $data
+     */
+    private function createWatchdogFromArray(array $data): void
+    {
+        $product = $this->getReference(ProductDataFixture::PRODUCT_PREFIX . $data[self::ATTRIBUTE_PRODUCT_KEY]);
+
+        $watchdogData = $this->watchdogDataFactory->createByDomainId($data[self::ATTRIBUTE_DOMAIN_KEY]);
+        $watchdogData->email = $data[self::ATTRIBUTE_EMAIL_KEY];
+        $watchdogData->product = $product;
+        $watchdogData->validUntil = new DateTime('now' . ' + ' . random_int(2, 730) . ' days');
```

Comment:
I would consider add also a different creation time for the sake of testing

Reply:
I see the thumb, but not the change 😅  
Is it intentional?

Reply:
Yes, I forgot to send the comment. You mentioned in a different comment that changing DateTime to DateTimeImmutable isn't necessary, so I left it as it is here as well. Should I change the DateTime throughout the entire entity then?